ui/plugins/eu.esdihumboldt.hale.ui.functions.numeric/src/eu/esdihumboldt/hale/ui/functions/numeric/SequentialIDParameterPage.java

```diff
@@ -377,13 +377,22 @@ protected String generateExample() {
 		String suffix = (this.suffix == null) ? ("") : (this.suffix.getText());
 		String startValue = (this.startValue == null) ? ("") : (this.startValue.getText());
 
+		String specialCharactersString = "!@#$%&*()'+,-./:;<=>?[]^_`{|}";
+
+		for (int i = 0; i < startValue.length(); i++) {
+			char ch = startValue.charAt(i);
+			if (specialCharactersString.contains(Character.toString(ch))
+					|| startValue.startsWith("0")) {
+				return "StartValue cannot have leading 0s or contain special characters";
```

Comment:
1. What about alphabets. Are they allowed? are decimals allowed? confirm it first from Thorsten or Kate.
A cleaner way would be to use pattern matcher refer
- https://stackoverflow.com/a/15111450 or
- https://www.geeksforgeeks.org/how-to-check-if-string-contains-only-digits-in-java/ -- point `3 Using Regular Expression:`

2. When you return an error message are you sure the finish button is not enabled?

Reply:
I know there are two ways to check for forbidden chars, with or without regex. I have decided without it because I find it more intuitive. Can it stay?

Reply:
> I know there are two ways to check for forbidden chars, with or without regex. I have decided without it because I find it more intuitive. Can it stay?

I know this approach is more intuitive and you can clearly see a list of things that are excluded. Also the complexity of this approach and with the one using Patternmatcher is same O(N). However, there are drawbacks of this approach:
- in the line `char ch = startValue.charAt(i);` you are first getting a character from a string and then converting that character back to string with `Character.toString(ch)` to check if it is present in the special characters string. So, this code looks a little extra effort.
- You have to manually add all the special characters that you want to exclude. What if you miss one? For instance you missed `\`.  Debugging the missed one would be a nightmare. 
- If later you decide to exclude alphabets then you would need to add `A, B, C...upto Z` and `a, b, c.... upto z`. or else you need to add some other condition with `||` to validate for alphabets. That will make your code look clumsy.
- So your if condition will look something like this: 

```
if (specialCharactersString.contains(Character.toString(ch)) || startValue.startsWith("0") || Character.isAlphabetic(startValue.tolowercase()))
```
					
But still if you prefer to use the one without regex then it is absolutely fine.

Reply:
What's the goal here? If it is to check whether the input can be parsed as an integer, it might be easier to just `try` to parse it and do the error handling in the `catch` block.

Regarding the leading zeros: I'd suggest to implement the example generation in a way so that a check for leading zeros in not needed, e.g. by using the parsed `Integer` value of the input.

Also, shouldn't `PARAM_START_VALUE` be modelled as an `Integer` instead of a `String`?